pkg/controllers/internalmembercluster/v1beta1/member_controller.go

```diff
@@ -156,20 +272,183 @@ func (r *Reconciler) stopAgents(ctx context.Context, imc *clusterv1beta1.Interna
 
 // updateHealth collects and updates member cluster resource stats and set ConditionTypeInternalMemberClusterHealth.
 func (r *Reconciler) updateHealth(ctx context.Context, imc *clusterv1beta1.InternalMemberCluster) error {
-	klog.V(2).InfoS("updateHealth", "InternalMemberCluster", klog.KObj(imc))
-
-	if err := r.updateResourceStats(ctx, imc); err != nil {
-		r.markInternalMemberClusterUnhealthy(imc, fmt.Errorf("failed to update resource stats %s: %w", klog.KObj(imc), err))
+	klog.V(2).InfoS("Updating health status", "InternalMemberCluster", klog.KObj(imc))
+
+	probeRes := r.rawMemberClientSet.Discovery().RESTClient().Get().AbsPath(healthProbePath).Do(ctx)
```

Comment:
Yeah, this is probing the API server at the health endpoint. For other components it might be better to explore later as at this moment the health condition we add is largely unused. The pressure should be light.